### ISSUE TITLE
Relational expansion NEEDS a parameter

### ISSUE THREAD

**Issue by geferon** [PlaceholderAPI/PlaceholderAPI#941]
### Confirmation

- [X] My issue isn't already found on the Issue tracker.
- [X] My issue is about **PlaceholderAPI** and not any expansion or external plugin
- [X] The issue isn't already fixed in a Spigot Release or Development Build.
- [X] The [Common Issues](https://github.com/PlaceholderAPI/PlaceholderAPI/wiki/Common-Issues) page doesn't mention this issue.

### Type

API Bug

### What happens?

It seems like creating a relational expansion requires you to have an underscore to be able to be ran.

Mainly because of this check here.
https://github.com/PlaceholderAPI/PlaceholderAPI/blob/3fca78626a024ea5c2dc6702a4efcf00d0dc30ed/src/main/java/me/clip/placeholderapi/PlaceholderAPI.java#L185

### Expected Behaviour

I wanted to register a simple placeholder expansion that calculates the distance between two players, so I registered one named "distance", and tried referencing it via "%rel_distance%", whereas to my frustration, it wasn't working.
I took a look at the code, saw that condition, and replaced it with "%rel_distance_%", where it started to work.
This should not be the case, as most other placeholders can be used without the need for parameters/identifiers.

### How to Reproduce

```java
public class DistancePlaceholder extends PlaceholderExpansion implements Relational {
    @Override
    public @NotNull String getIdentifier() {
        return "distance";
    }

    @Override
    public @NotNull String getAuthor() {
        return "Geferon";
    }

    @Override
    public @NotNull String getVersion() {
        return "1.0.0"
    }

    @Override
    public String onPlaceholderRequest(Player one, Player two, String identifier) {
        if (one == null || two == null)
            return null;

        if (one.getLocation().getWorld() != two.getLocation().getWorld())
            return null;

        int distance = (int) Math.round(one.getLocation().distance(two.getLocation()));
        return Integer.toString(distance);
    }
}
```

Expected behaviour:
`%rel_distance%` should return the distance in blocks between players.
Actual behaviour:
This placeholder only works if you use it like so: `%rel_distance_%`

### `/papi dump` Output

https://paste.helpch.at/dufipihiwi

### Additional Info

Code is already present in the "How to Reproduce".

**Comment by Andre601** [comment]
I believe this to be intentional behaviour.
Normal placeholders have a `%<identifier>_<values>%` format, and relational ones are no different to this with the only change being prefixed with `rel_`.

So your issue is essentially that you're not providing any basic value for PlaceholderAPI to recognize.

**Comment by geferon** [comment]
Thing is, normal placeholders DO work without values, so if this weren't a relational placeholder, and you were to use `%distance%`, it works, whereas if you don't use the values part in the relational placeholder, it DOESN'T work, which makes it inconsistant.

**Comment by Andre601** [comment]
*normal* placeholders are `%<identifer>_value%` (i.e. `%player_name%`), so if one like `%player%` works is it not PlaceholderAPI but whatever plugin you use the placeholder in that parses it.

The parsing logic acts like that for all placeholders and relational ones just have the difference of the afforementioned `rel_` prefix and that it uses 2 players instead of just one.

**Comment by BlitzOffline** [comment]
Andre is wrong this time. While I agree with him that it shouldn't work if it doesn't have at least one `_` to split the identifier from the rest of the arguments, it is currently inconsistent and normal placeholders work without it. I've reported this before in discord but my message got lost to time and other conversations.

**Comment by Andre601** [comment]
Yeah, my bad. Never noticed that the charsreplacer accepts a `%identifier%` pattern as valid placeholder.

To be honest, the relational placeholder system seems a bit dated in my opinion as the `rel_` requirement is weird...
Perhaps merging it and making the CharsReplacer check for if the expansion is a relational one could be useful here to do and get rid of that `rel_` prefix requirement...
Tho, at the same time do I fear that it would break existing placeholder expansions that may offer normal and relational ones, as it would then prioritize the relational ones.

**Comment by geferon** [comment]
> Andre is wrong this time. While I agree with him that it shouldn't work if it doesn't have at least one `_` to split the identifier from the rest of the arguments, it is currently inconsistent and normal placeholders work without it. I've reported this before in discord but my message got lost to time and other conversations.

I don't mind either or, it's just that what would be nice would be for this to be specified in the wiki and also made consistent.
